File: lib/features/auth/domain/entities/user_entity.dart
class UserEntity {
  final String id;
  final String name;
  final String email;
  final String role;
  final String? avatar;
  final bool isActive;
  final bool acceptedTerms;
  final String? subscriptionPlan;
  final String? subscriptionStatus;

  const UserEntity({
    required this.id,
    required this.name,
    required this.email,
    required this.role,
    this.avatar,
    required this.isActive,
    this.acceptedTerms = false,
    this.subscriptionPlan,
    this.subscriptionStatus,
  });

  /// true si tiene cualquier plan de pago activo (premium_user o agency)
  bool get isPremium => subscriptionPlan != null && subscriptionStatus == 'active';
}

File: backend/app/core/domain/entities/user_entity.py
from dataclasses import dataclass
from datetime import datetime
from typing import Optional
from enum import Enum

class UserRole(str, Enum):
    USER = "USER"
    SELLER = "SELLER"
    AGENCY = "AGENCY"
    ADMIN = "ADMIN"

@dataclass
class UserEntity:
    id: str
    name: str
    email: str
    role: UserRole
    password_hash: Optional[str] = None
    avatar: Optional[str] = None
    is_active: bool = True
    accepted_terms: bool = False
    created_at: Optional[datetime] = None

    def can_publish(self) -> bool:
        return self.role in [UserRole.SELLER, UserRole.AGENCY, UserRole.ADMIN]

    def is_admin(self) -> bool:
        return self.role == UserRole.ADMIN
